cloudsync/api_test.py

```diff
@@ -1,25 +1,45 @@
 """
 Tests for api
 """
-
-from io import BytesIO
+import os
+import io
 
 import boto3
 import pytest
+from botocore.exceptions import ClientError
 from django.conf import settings
+from django.contrib.auth.models import User
+from django.db import transaction
+from django.test import override_settings
 from moto import mock_s3
 
-from cloudsync.api import process_transcode_results, refresh_status
+from cloudsync.api import process_transcode_results, refresh_status, extract_title, process_watch_file, transcode_video
 from cloudsync.conftest import MockClientET, MockBoto
+from cloudsync.utils import VideoFilenameError
 from ui.constants import VideoStatus
 from ui.factories import (
     VideoFactory,
     EncodeJobFactory,
     VideoFileFactory,
-)
+    UserFactory)
+from ui.models import Video
 
 pytestmark = pytest.mark.django_db
 
+# pylint: disable=redefined-outer-name
+
+
+@pytest.fixture()
+def video():
+    """Fixture to create a video"""
+    return VideoFactory()
+
+
+@pytest.fixture()
+def videofile(video):
+    """Fixture to create a videofile"""
+    return VideoFileFactory(video=video)
```

Comment:
I do not think this is a good approach with the fixtures: the `VideoFileFactory` takes care of creating a video if not passed; unless I misunderstood what you are trying to do, there is not need for passing the video fixture as parameter here.